application-task-manager-default/src/test/java/com/xwiki/taskmanager/TaskMacroUpdateEventListenerTests.java

```diff
@@ -20,171 +20,181 @@
 
-        this.listener.onEvent(null, document, context);
-
-        verify(document, never()).newXObject(TaskMacroUpdateEventListener.TASK_CLASS_REFERENCE, context);
-//        verify(obj1).set(Task.ASSIGNEES, Arrays.asList("XWiki.Admin", "XWiki.User1"), context);
     }
 
     @Test
-    public void createMacroObjectTest() throws XWikiException, ParseException
+    public void onDeletingEventTest() throws TaskException
     {
+        this.eventListener.onEvent(new DocumentDeletingEvent(), this.docWithTasks, this.context);
 
-//        when(this.document.getXObjects(TaskMacroUpdateEventListener.TASK_CLASS_REFERENCE)).thenReturn(Collections.emptyList());
-//        when(this.document.newXObject(TaskMacroUpdateEventListener.TASK_CLASS_REFERENCE, context)).thenReturn(obj1);
-
-        this.listener.onEvent(null, document, context);
-
-//        verify(document).newXObject(TaskMacroUpdateEventListener.TASK_CLASS_REFERENCE, context);
-//
-//        verify(obj1).set(Task.CREATOR, "XWiki.Admin", context);
-//        verify(obj1).set(Task.STATUS, task1.isCompleted(), context);
-//        verify(obj1).set(Task.CREATE_DATE, task1.getCreateDate(), context);
-//        verify(obj1).set(Task.DESCRIPTION, task1.getDescription(), context);
-//
-//        verify(obj1).set(Task.ASSIGNEES, Collections.singletonList("XWiki.Admin"), context);
-//        verify(obj1).set(Task.DEADLINE, task1.getDeadline(), context);
-//        verify(obj1).set(Task.DESCRIPTION, task1.getDescription(), context);
+        verify(this.taskManager).deleteTasksByOwner(this.pageWithMacro);
     }
 
     @Test
-    public void removeMacroObjectWhenThereIsNoCorrespondingMacroOnPageTest() {
-        when(this.taskXDOMProcessor.extract(any(), any())).thenReturn(Collections.emptyList());
-
-        this.listener.onEvent(null, document, context);
-
-//        verify(this.document).removeXObject(obj1);
-        verify(this.document, never()).setTitle(any());
+    public void onUpdatingWithRemovedTaskEventTest() throws XWikiException
+    {
+        when(this.taskXDOMProcessor.extract(this.docXDOM, this.pageWithMacro)).thenReturn(new ArrayList<>(Collections.singletonList(task)));
+        when(this.taskXDOMProcessor.extract(this.prevVersionDocXDOM, this.pageWithMacro)).thenReturn(new ArrayList<>(Collections.singletonList(task_1)));
+        when(this.taskDoc.isNew()).thenReturn(true);
+        when(this.authorizationManager.hasAccess(Right.EDIT, taskPage)).thenReturn(true);
+        when(this.authorizationManager.hasAccess(Right.DELETE, taskPage_1)).thenReturn(true);
+
+        this.eventListener.onEvent(new DocumentUpdatingEvent(), this.docWithTasks, this.context);
+
+        verify(this.wiki).getDocument(this.taskPage, this.context);
+        verify(this.taskObj).set(Task.OWNER, this.pageWithMacro, this.context);
+        verify(this.authorizationManager).hasAccess(Right.EDIT, taskPage);
+        verify(this.authorizationManager).hasAccess(Right.DELETE, taskPage_1);
```

Comment:
Same as above, you should rely on ``when()`` for these methods in order to control what they return. The execution is different based on what they return (i.e. whether the current user has edit right or not).